### ISSUE TITLE
The right way to setup multisite (one codebase, multiple DBs)?

### ISSUE THREAD

**Issue by haixiang** [docksal/docksal#199]
**Description**

I am trying to setup multisite in docksal. I have shared codebase and multiple DBs. I can create separate init scripts for each site, but can't find a straightforward way to have different settings in docksal.env file: VIRTUAL_HOST, SITE_DIRECTORY and MYSQL_DATABASE specifically.

**Describe the results you expected:**

Either different sections in docksal.env file to allow sets of settings or separate docksal.[site n].env files?

**Comment by achekulaev** [comment]
`VIRTUAL_HOST` supports Drupal multi-site out of the box if you used fin config generate or use zero-configuration. If you have configuration generated your `docksal.yml` should have this line:

```yml
web:
  ...
  labels:
    - io.docksal.virtual-host=${VIRTUAL_HOST},*.${VIRTUAL_HOST}
    ...
```

So your `VIRTUAL_HOST` variable is supposed to **not** have wildcard in it. Just domain is enough. 
```
VIRTUAL_HOST=mysite.docksal
```

Sub-domains routed automatically by that above mentioned line in yml. So `VIRTUAL_HOST` is already covered. You don't need to manually add each sub-site.

`SITE_DIRECTORY` is a custom variable, you may customize this variable and init script as you wish. There is no bindings to this variable inside fin. Drupal 7 or 8 site that you probably used as an example, are just examples. Their init scripts are not built to support all possible use cases.

`MYSQL_DATABASE` sets the **default** database. Though you can use multiple databases, there can only be 1 default database. For MySQL related commands to work with other databases you need to specify database name via `--db=...` parameter. See help for those commands (e.g. `fin help sqli`)

If your have more questions please ask. If you feel your questions were answered please close the ticket.

**Comment by haixiang** [comment]
Thank you, that helps!

**Comment by sharique** [comment]
What if we want site in parallel like site1.docksal and site2.docksal

**Comment by lmakarov** [comment]
@sharique I assume your question is in regards to multiple custom domains. It's covered in docs here:
https://docs.docksal.io/core/system-vhost-proxy/#using-arbitrary-custom-domains

**Comment by phoang11** [comment]
@lmakarov I want to setup for one code base for multi-sites and multi dbs.

code base
  |--site 1 --> site1.docksal -> site1 db
  |--site 2 --> site2.docksal -> site2 db

Any insights for me to setup that?

**Comment by lpeabody** [comment]
@phoang11 If you reference the documentation here (https://docs.docksal.io/core/system-vhost-proxy/#using-arbitrary-custom-domains) it shows you how to add non-default domains to your project. So you really just need to follow those instructions and then update your sites.php to map the domains to the proper site folder.

**Comment by lmakarov** [comment]
Since this has been brought up a few times recently, I created a new issues to cover this use case in the docs: https://github.com/docksal/docksal/issues/1208
